### ISSUE TITLE
Item 31. 한정적 와일드카드를 사용해 API 유연성을 높이라.

### ISSUE THREAD

**Issue by youngkimi** [peaches-book-study/effective-java#30]
### Assignee : youngkimi

---

# 🍑 서론

# 🍑 본론

# 🍑 결론

---

## Referenced by

-

**Comment by jseok0917** [comment]
네, 클래스를 만들 때에도 와일드카드(?)를 사용할 수 있습니다. 와일드카드는 제네릭 타입에서 사용되며, 타입 파라미터를 알 수 없는 경우에 유용하게 활용됩니다.

와일드카드는 보통 제네릭 클래스, 제네릭 메서드, 제네릭 인터페이스의 타입 파라미터로 사용됩니다. 와일드카드는 다음과 같은 세 가지 형태로 사용될 수 있습니다:

<?>: 모든 타입을 나타냅니다. 제네릭 클래스나 메서드에 와일드카드를 사용하면 해당 위치에 어떤 타입이든 사용할 수 있습니다.

<? extends T>: 상한 경계 와일드카드입니다. T의 하위 클래스들을 나타냅니다. 따라서 T 자신이나 T의 하위 클래스의 인스턴스만 사용할 수 있습니다.

<? super T>: 하한 경계 와일드카드입니다. T의 상위 클래스들을 나타냅니다. 따라서 T 자신이나 T의 상위 클래스의 인스턴스만 사용할 수 있습니다.

와일드카드는 제네릭 타입을 더 유연하게 만들어주며, 다양한 상황에서 유용하게 사용될 수 있습니다.
